### ISSUE TITLE
ACF Global options and Menus

### ISSUE THREAD

**Issue by ccorda** [patronage/bubs-next#47]
We often define a series of user editable globals in WordPress that a user can customize:

* Header/Footer menu items
* Social meta tags
* Global strings of text

I'm not sure we want to load these on every page -- they change infrequently. 

Can we merge these options in with our constants file on build time, and then use a webhook to trigger a new build when they change?

Or should we grab these options and commit them to git for preview and approval before changing? One thing I like about this approach is that it would give us the ability to review something like a menu change to production, which might possibly break things.

**Comment by ccorda** [comment]
Looked into this a bit more, and this discussion thread touches on global data:
https://github.com/vercel/next.js/discussions/10949

It looks like you can't get it in `_app.js`, but we can add to WordPress and make a `getGlobals()` that can be used in next only pages, and that graphql can merge in that fragment so that we still have a single API call to WordPress: https://github.com/vercel/next.js/discussions/10949#discussioncomment-80171

**Comment by ccorda** [comment]
Some updates from various projects that would be my recommended path forward:

* Using a menu fragment, and named menus to grab menus by location. This is most reliable since sometimes different menus have different IDs in different environments (or if a user changes):

``` js
export async function getGlobalProps() {
  let query = /* GraphQL */ `
  
    fragment GlobalOptions on RootQuery {
      themeSettings
    }
    
    fragment Menus on RootQuery {
      menuHeader: menus(where: { location: HEADER }) {
        ${fragmentMenu}
      }
      menuFooter: menus(where: { location: FOOTER }) {
        ${fragmentMenu}
      }
    }

    query AllGlobals {
      ...Menus
      ...GlobalOptions
    }
  `;

  const data = await fetchAPI(query);
  return data;
}
```

* Saving in context and accessing via `GlobalsProvider`.
* Helper in headers/footer to output either single or two tier menus
* Mobile version with `FaBars` and simple full screen slidedown. (and how do we manage global z-index?)
* Desktop with js-less dropdown from Bootstrap

Having all these in place will be a significant time saver.

**Comment by kylehotchkiss** [comment]
What are the next steps for this ticket? Is the global props code loaded in this repo yet?

**Comment by ccorda** [comment]
https://github.com/patronage/bubs-next/pull/114

**Comment by kylehotchkiss** [comment]
#114 is merged and I have an open PR for #139 to demonstrate a working global variable set via theme settings options page. This one seems safe to close out.
